framework/src/actions/CreateApplicationBlockAction.C

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreateApplicationBlockAction.h"
+#include "MooseApp.h"
+#include "MooseObjectAction.h"
+#include "ActionFactory.h"
+#include "MooseUtils.h"
+
+registerMooseAction("MooseApp", CreateApplicationBlockAction, "create_application_block");
+
+InputParameters
+CreateApplicationBlockAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addParam<std::string>("type", "MooseTestApp", "The name of the app needs to create.");
```

Comment:
```suggestion
  params.addParam<std::string>("type", "MooseTestApp", "The name of the app that should run this input file.");
```

Reply:
not sure about this default.

I think we would want to default to whatever the app is

Reply:
I can leave it empty here. The apptype is actually set before this action